src/hal/src/command/mod.rs

```diff
@@ -65,100 +64,131 @@ impl Level for Primary { }
 pub enum Secondary { }
 impl Level for Secondary { }
 
-/// Thread-safe finished command buffer for submission.
-pub struct Submit<B: Backend, C, S, L>(pub(crate) B::CommandBuffer, pub(crate) PhantomData<(C, S, L)>);
-impl<B: Backend, C, S, L> Submit<B, C, S, L> {
-    fn new(buffer: B::CommandBuffer) -> Self {
-        Submit(buffer, PhantomData)
-    }
+/// A property of a command buffer to be submitted to a queue.
+pub trait RawSubmittable<B: Backend> {
```

Comment:
I think this should be implemented for `B::CommandBuffer` at least.
Otherwise user woudln't be able to submit raw `&B::CommandBuffer` into raw `B::CommandQueue` which seems odd :)

Maybe `Borrow<B::CommandBuffer>` would make more sense than `RawSubmittable`